modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/GeoJSONReader.java

```diff
@@ -86,6 +86,8 @@ public class GeoJSONReader implements AutoCloseable {
 
     private SimpleFeatureBuilder builder;
 
+    private static final String ID = "id";
```

Comment:
The user should be able to choose this value, it could be `fid` or `gid` etc

Reply:
The standard is pretty specific about this:
If a Feature has a commonly used identifier, that identifier
      SHOULD be included as a member of the Feature object with the name
      "id", and the value of this member is either a JSON string or
      number.

Reply:
Yes, but SHOULD is not mandatory, see the RFC titled "[Key words for use in RFCs to Indicate Requirement Levels](https://datatracker.ietf.org/doc/html/rfc2119)", quoting:

```
SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.
```

So strong recommendation, but not an absolute requirement. Also, the wording talks about identifier, but does not say anything about its uniqueness and in which context (e.g., in the STAC spec the id must be unique only inside that collection). To make another example, I identify as Andrea, but it's too common, Andrea Aime is better, it's (accidentaly) unique in the province where I live, but if I want uniqueness at the country level, I have to switch to the SSN.

Reply:
Gotcha.  Updated to add constructor that allows specification of an id field name.